FirebaseStorageSwift/Tests/Integration/StorageAsyncAwait.swift

```diff
@@ -0,0 +1,391 @@
+      let ref = storage.reference(withPath: "ios/public/1mb2")
+
+      let metadata = StorageMetadata()
+      metadata.cacheControl = "cache-control"
+      metadata.contentDisposition = "content-disposition"
+      metadata.contentEncoding = "gzip"
+      metadata.contentLanguage = "de"
+      metadata.contentType = "content-type-a"
+      metadata.customMetadata = ["a": "b"]
+
+      let updatedMetadata = try await ref.updateMetadata(metadata)
+      assertMetadata(actualMetadata: updatedMetadata,
+                     expectedContentType: "content-type-a",
+                     expectedCustomMetadata: ["a": "b"])
+
+      let metadata2 = updatedMetadata
+      metadata2.contentType = "content-type-b"
+      metadata2.customMetadata = ["a": "b", "c": "d"]
+
+      let metadata3 = try await ref.updateMetadata(metadata2)
+      assertMetadata(actualMetadata: metadata3,
+                     expectedContentType: "content-type-b",
+                     expectedCustomMetadata: ["a": "b", "c": "d"])
+      metadata.cacheControl = nil
+      metadata.contentDisposition = nil
+      metadata.contentEncoding = nil
+      metadata.contentLanguage = nil
+      metadata.contentType = nil
+      metadata.customMetadata = nil
+      let metadata4 = try await ref.updateMetadata(metadata)
+      XCTAssertNotNil(metadata4)
+    }
+
+    func testPagedListFiles() async throws {
+      let ref = storage.reference(withPath: "ios/public/list")
+      let listResult = try await ref.list(maxResults: 2)
+      XCTAssertEqual(listResult.items, [ref.child("a"), ref.child("b")])
+      XCTAssertEqual(listResult.prefixes, [])
+      guard let pageToken = listResult.pageToken else {
+        XCTFail("pageToken should not be nil")
+        return
+      }
+      let listResult2 = try await ref.list(maxResults: 2, pageToken: pageToken)
+      XCTAssertEqual(listResult2.items, [])
+      XCTAssertEqual(listResult2.prefixes, [ref.child("prefix")])
+      XCTAssertNil(listResult2.pageToken, "pageToken should be nil")
+    }
+
+    func testListAllFiles() async throws {
+      let ref = storage.reference(withPath: "ios/public/list")
+      let listResult = try await ref.listAll()
+      XCTAssertEqual(listResult.items, [ref.child("a"), ref.child("b")])
+      XCTAssertEqual(listResult.prefixes, [ref.child("prefix")])
+      XCTAssertNil(listResult.pageToken, "pageToken should be nil")
+    }
+
+    private func waitForExpectations() {
+      let kFIRStorageIntegrationTestTimeout = 60.0
+      waitForExpectations(timeout: kFIRStorageIntegrationTestTimeout,
+                          handler: { (error) -> Void in
+                            if let error = error {
+                              print(error)
+                            }
+                          })
+    }
+
+    private func assertResultSuccess<T>(_ result: Result<T, Error>,
+                                        file: StaticString = #file, line: UInt = #line) {
```

Comment:
I believe this helper can be removed if `testSimplePutFile` in line 109 is removed.

Reply:
yep!